packages/pubsub/src/messages/attachDocMessage.ts

```diff
@@ -0,0 +1,22 @@
+import { PubsubMessage } from './pubsubMessage';
+
+export const attachDocMessageType = 'attachDoc';
+
+type AccessDataPayload = {
+    id: string;
+    rights: ('read' | 'write')[];
+};
+
+export type AttachDocPayload = {
+    docs: string[];
+    connection: {
+        id: string;
+        tokenId: string;
+        tokenExpiredAt: Date;
+        userId: string;
+        consumerId: string;
+        access: Record<string, AccessDataPayload>;
+    } & WebSocket;
```

Comment:
We need to pass a `WebSocket` object for DocService to use it. `DocService` creates DocManager and adds the connection(`WebSocket` ojbect) to the `DocManger`, so it can send changes to this connection. 

This is required to reattach the connection if a doc read rights are revoked and granted again while a connection is still active. One more case is: a consumer has updated a token with access to a new document. In such case we need to attach an existing connection to the document.

The attach message calls joinToDoc method
```ts
    private onAttachDocMessage = (message: AttachDocPubsubMessage) => {
        for (const docId of message.message.docs) {
            this.joinToDoc(message.message.connection, docId);
        }
    };
```
https://github.com/a-la-fois/a-la-fois/blob/connected_permissions/packages/messageProxy/src/doc/doc.service.ts#L93-L117